### ISSUE TITLE
Patch fails to apply to RT 5.0.4

### ISSUE THREAD

**Issue by PinkFreud** [gibus/RT-Extension-ConditionalCustomFields#12]
[This commit](https://github.com/bestpractical/rt/commit/bcff787228b844ad9bde4091bc622c973335fb5d#diff-348d766ffe1d809bf3867fcbd05240e3fa4fa7d956ebdd1a01f798aa0f9e0bbc), from last Aug, breaks 5.0-add-callbacks-to-extend-customfields-capabilities.patch - while it occurred too late for the 5.0.3 release, it was included in last week's 5.0.4 update.

As the first two files had the patch apply with fuzzing, I'm attaching a new patch which appears to work with 5.0.4.

[5.0.4-add-callbacks-to-extend-customfields-capabilities.zip](https://github.com/gibus/RT-Extension-ConditionalCustomFields/files/11425211/5.0.4-add-callbacks-to-extend-customfields-capabilities.zip) - my apologies for zipping this, but github appears to have broken uncompressed patch attachments.

**Comment by PinkFreud** [comment]
Actually, scratch that - navigating to Admin->Queues->Create causes RT to throw an error:

`An internal RT error has occurred. Your administrator can find more details in RT's log files.`

```
[365898] [Mon May  8 22:01:36 2023] [error]: Error during compilation of /data1/opt/rt5/share/html/Admin/Queues/Modify.html:
Global symbol "$CF" requires explicit package name (did you forget to declare "my $CF"?) at /data1/opt/rt5/share/html/Admin/Queues/Modify.html line 174.


Stack:
  [/usr/local/share/perl5/5.32/HTML/Mason/Interp.pm:450]
  [/usr/local/share/perl5/5.32/HTML/Mason/Request.pm:250]
  [/usr/local/share/perl5/5.32/HTML/Mason/Request.pm:212]
  [/data1/opt/rt5/sbin/../lib/RT/Interface/Web/Request.pm:63]
  [/usr/local/share/perl5/5.32/Class/Container.pm:275]
  [/usr/local/share/perl5/5.32/Class/Container.pm:353]
  [/usr/local/share/perl5/5.32/HTML/Mason/Interp.pm:348]
  [/usr/local/share/perl5/5.32/HTML/Mason/Interp.pm:342]
  [/usr/local/share/perl5/5.32/HTML/Mason/PSGIHandler.pm:59]
  [/usr/local/share/perl5/5.32/HTML/Mason/PSGIHandler/Streamy.pm:52]
  [/usr/local/share/perl5/5.32/Plack/Util.pm:339]
  [/usr/local/share/perl5/5.32/Plack/Util.pm:339]
  [/usr/local/share/perl5/5.32/Plack/Handler/FCGI.pm:149]
  [/usr/local/share/perl5/5.32/Plack/Loader.pm:84]
  [/usr/local/share/perl5/5.32/Plack/Runner.pm:279]
  [/data1/opt/rt5/sbin/../lib/RT/PlackRunner.pm:150]
  [/opt/rt5/sbin/rt-server.fcgi:162] (/data1/opt/rt5/sbin/../lib/RT/Interface/Web/Handler.pm:216)
```

**Comment by gibus** [comment]
Hello,
I haven't tested yet with RT 5.0.4. But looking at your proposed patch and file `rt5/share/html/Admin/Queues/Modify.html` for RT 5.0.4, it seems you've applied the following code in the wrong place:

```
diff --git a/share/html/Admin/Queues/Modify.html b/share/html/Admin/Queues/Modify.html
index f474102..f320bf9 100644
--- a/share/html/Admin/Queues/Modify.html
+++ b/share/html/Admin/Queues/Modify.html
@@ -171,6 +171,7 @@
       </&>
 % }
     </div>
+% $m->callback( CallbackName => 'AfterCustomFieldValue', CustomField => $CF, Object => $QueueObj);
   </div>
 
   <div class="form-row">
```

it should be:

```
diff --git a/share/html/Admin/Queues/Modify.html b/share/html/Admin/Queues/Modify.html
index f47410290..35522ba9e 100644
--- a/share/html/Admin/Queues/Modify.html
+++ b/share/html/Admin/Queues/Modify.html
@@ -121,6 +121,7 @@
       <& /Elements/EditCustomField, CustomField => $CF,
                                     Object => $QueueObj, &>
 </&>
+% $m->callback( CallbackName => 'AfterCustomFieldValue', CustomField => $CF, Object => $QueueObj);
 % }
 
 % if ( RT->Config->Get('Crypt')->{'Enable'} ) {
```

But it's only by reading the code, I haven't tested it…

**Comment by PinkFreud** [comment]
That's odd - that's not the one I had to fix.  That patch appears to have applied with fuzzing, so I assumed the line was in the right place.

I can try moving that line up there, though, to see if that fixes the issue.

**Comment by PinkFreud** [comment]
This appears to work now, with your above suggestion.  Attaching final patch for 5.0.4 here.

[5.0.4-add-callbacks-to-extend-customfields-capabilities-20230509.zip](https://github.com/gibus/RT-Extension-ConditionalCustomFields/files/11436705/5.0.4-add-callbacks-to-extend-customfields-capabilities-20230509.zip)

**Comment by metbosch** [comment]
Is the patch for 5.0.4 available in the master branch? Or should it work with the previous one?

**Comment by gibus** [comment]
Sorry I've just committed the patch to master branch and released v1.09 which should be available on metacpan in a few minutes
